### PR TITLE
[DEVOPS-463] add a simpler API for hash-installer

### PR DESCRIPTION
**Pull request by cleverca22**

### CODE REVIEW

**Comment by jmitchell** [comment]
@cleverca22: I recommend enabling stylish haskell as explained at https://github.com/input-output-hk/cardano-sl-style-guides/blob/master/haskell-style-guide.md

File: auxx/src/HashInstaller.hs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Main where
+
+import qualified Data.ByteString.Lazy as BSL
+import           Formatting           (sformat, string, (%))
+import           Pos.Crypto           (Hash, SignTag (SignUSVote),
+                                       emptyPassphrase, encToPublic, hash,
+                                       hashHexF, safeSign, unsafeHash,
+                                       withSafeSigner, withSafeSigners)
```
Comment (deepfire) [inline auxx/src/HashInstaller.hs]: Most of those imports are not require, I guess..
Also maintaining them is a pain, so just using qualified names might help.

Comment (cleverca22) [reply auxx/src/HashInstaller.hs]: i'm rebuilding with `-Wall` to see what is extra

File: auxx/src/HashInstaller.hs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Main where
+
+import qualified Data.ByteString.Lazy as BSL
+import           Formatting           (sformat)
+import           Pos.Crypto               (hashHexF)
```
Comment (jmitchell) [inline auxx/src/HashInstaller.hs]: Why did the spacing end up weird here even after using stylish? Maybe there's a tab.

Comment (cleverca22) [reply auxx/src/HashInstaller.hs]: i think thats a refactor i did after stylish, re-running it

**Comment by jagajaga** [comment]
It should be in `auxx`.

**Comment by cleverca22** [comment]
auxx is overly strict about having every parameter present and valid, even ones that have zero impact on the hash
`cardano-auxx cmd --commands "hash-installer Daedalus-installer-1.0.3253.pkg" --system-start 0`
with the requirement that `node/configuration.yaml` exists relative to the working directory, or you also specified `--configuration-file`
vs
`cardano-hash-installer Daedalus-installer-1.0.3253.pkg` with no extra flags and no expectations on other files being present

**Comment by georgeee** [comment]
@cleverca22 anyway it's not good to duplicate code and create many single-purpose executables. It's not very hard to adjust `cardano-auxx` according to your request (to have less params) and this way should be preferred

**Comment by cleverca22** [comment]
yeah, that will also work

**Pull request merged**